.github/workflows/_create_prepare_aks_cluster.yml

```diff
@@ -0,0 +1,119 @@
+name: Create and Prepare AKS Cluster
+
+on:
+  workflow_call:
+    outputs:
+      cluster-name:
+        description: AKS Cluster Name
+        value: ${{ jobs.create-prepare-aks-cluster.outputs.cluster-name }}
+  workflow_dispatch:
+    
+jobs:
+  create-prepare-aks-cluster:
+    name: Create and Prepare Test AKS Cluster
+    runs-on: ubuntu-latest
+    env: 
+      RESOURCE_GROUP: ${{ vars.RESOURCE_GROUP }} 
+      USER_ASSIGNED_IDENTITY_NAME: ${{ vars.USER_ASSIGNED_IDENTITY_NAME }}
+    outputs:
+      cluster-name: ${{ steps.create-aks-cluster.outputs.cluster-name }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get Workflow ID
+        id: generate-id
+        run: echo "id=$(openssl rand -hex 8 | tr -d '\n')" >> $GITHUB_OUTPUT
+      
+      - name: Log into Azure
+        run: |
+          az login --service-principal \
+          --username ${{ secrets.AZURE_SERVICE_PRINCIPAL_APP_ID }} \
+          --password ${{ secrets.AZURE_SERVICE_PRINCIPAL_PASSWORD }} \
+          --tenant ${{ secrets.AZURE_SERVICE_PRINCIPAL_TENANT }}
+
+      - name: Install Dependencies 
+        id: install-dependencies 
+        run: |
+          result=$(az extension list -o table  2>&1 || true)
+          if [[ $result == *"aks-preview"* ]]; then
+            echo "aks-preview already installed, upgrading aks-preview version."
+            az extension update --name aks-preview
+          else
+            echo "aks-preview extension not found. Installing aks-preview..."
+            az extension add --name aks-preview
+          fi
+
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"  --fail-with-body
+          chmod +x kubectl
+          sudo mv kubectl /usr/local/bin
+          kubectl version 2>&1 || true
+
+      - name: Create AKS Cluster
+        id: create-aks-cluster
+        if: steps.install-dependencies.outcome == 'success'
+        env: 
+          RUN_ID: ${{ steps.generate-id.outputs.id}}
+        run: |
+          az aks create --resource-group ${{ vars.RESOURCE_GROUP }} \
+          --name skr-kafka-demo-${RUN_ID} \
+          --kubernetes-version ${{ vars.KUBERNETES_VERSION }} \
+          --os-sku AzureLinux \
+          --node-vm-size Standard_DC4as_cc_v5 \
+          --tags "Owner=accct" \
```

Comment:
I feel like the owner and the email should both be github secrets (the email especially to avoid spam).

Reply:
Only Owner tag is required, so I will remove other tags. The rest of the tags are already on the resource group.